tests/qnn/test_qnn_torch.py

```diff
@@ -418,6 +418,34 @@ def test_gradients(self, get_circuit, n_qubits):
             assert torch.allclose(g1, g2)
         assert len(weights) == len(list(layer.parameters()))
 
+    @pytest.mark.parametrize("n_qubits, output_dim", indices_up_to(1))
+    def test_cuda_backward(self):
+        if not torch.cuda.is_available():
+            pytest.skip("Cuda device not available")
+        else:
+            try:
+                n_qubits = 4
+                dev = qml.device("default.qubit", wires=n_qubits)
+
+                @qml.qnode(dev, interface="torch")
+                def circuit(inputs, weights):
+                    qml.templates.AmplitudeEmbedding(inputs, wires=range(n_qubits), pad=0.0)
+                    qml.templates.BasicEntanglerLayers(weights, wires=range(n_qubits))
+                    return [qml.expval(qml.PauliZ(wires=i)) for i in range(n_qubits)]
+
+                n_layers = 1
+                weight_shapes = {"weights": (n_layers, n_qubits)}
+
+                qlayer = qml.qnn.TorchLayer(circuit, weight_shapes)
+
+                x = torch.rand((5, 10))
+                norm = x.norm(p=2, dim=1, keepdim=True)
+                x_normalized = x.div(norm).to(torch.device("cuda"))
+                loss = torch.sum(qlayer(x_normalized)).squeeze()
```

Comment:
```suggestion
                x = torch.rand((5, n_qubits), dtype=torch.float64)
                loss = torch.sum(qlayer(x)).squeeze()
```

Reply:
Note that the new `x` definition has the `dtype` specified - we need to cast to `torch.float64` to match the `dtype` of the qnode. This is more of a quick fix here, we should look at having the `qlayer` return type being dynamic, either set by the user upon instantiation or updated when fed an `x` of a certain type.

Reply:
This could also be a side effect of my relatively old version of PyTorch.